src/partition/ReceivedPartition.cpp

```diff
@@ -610,10 +614,19 @@ void ReceivedPartition::createOwnerInformation()
   }
 }
 
-bool ReceivedPartition::areProvidedMeshesEmpty() const
+bool ReceivedPartition::isAnyProvidedMeshNonEmpty() const
 {
-  return (_fromMapping && not _fromMapping->getOutputMesh()->vertices().empty()) ||
-         (_toMapping && not _toMapping->getInputMesh()->vertices().empty());
+  for (mapping::PtrMapping fromMapping : _fromMappings) {
+    if (not fromMapping->getOutputMesh()->vertices().empty()) {
+      return true;
+    }
+  }
+  for (mapping::PtrMapping toMapping : _toMappings) {
+    if (not toMapping->getInputMesh()->vertices().empty()) {
+      return true;
```

Comment:
Do we need more tests here?

Reply:
Not sure what you mean :see_no_evil:

Reply:
In the diff view, codecov claims that this line is not covered.